server/etcdmain/config.go

```diff
@@ -115,9 +115,9 @@ func newConfig() *config {
 
 func (cfg *config) parse(arguments []string) error {
 	perr := cfg.cf.flagSet.Parse(arguments)
-	switch perr {
-	case nil:
-	case flag.ErrHelp:
+	switch {
+	case errors.Is(perr, nil):
```

Comment:
Think this has different semantics than checking `err == nil`. 
How this is an improvement?